File: benchmarks/fixtures/python-multi-hop-safe-path/src/storage.py
from pathlib import Path

DOCUMENT_ROOT = Path(__file__).resolve().parent.parent / "documents"
DOCUMENTS = {
    "welcome": DOCUMENT_ROOT / "welcome.txt",
}


def read_document(key):
    selected = DOCUMENTS.get(key)
    if selected is None:
        raise ValueError("unknown document")
    with open(selected, encoding="utf-8") as handle:
        return handle.read()

File: benchmarks/fixtures/javascript-multi-hop-safe-path/src/storage.js
import { readFile } from "node:fs/promises";
import { resolve } from "node:path";

const DOCUMENT_ROOT = resolve(import.meta.dirname, "..", "documents");
const DOCUMENTS = Object.freeze({
  welcome: resolve(DOCUMENT_ROOT, "welcome.txt"),
});

export function readDocument(key) {
  const selected = DOCUMENTS[key];
  if (selected === undefined) throw new Error("unknown document");
  return readFile(selected, "utf8");
}
